nilearn/decoding/tests/test_decoder.py

```diff
@@ -1211,33 +1203,37 @@ def test_decoder_adjust_screening_greaterthan_mask_surface(
 def test_decoder_fit_surface(decoder, _make_surface_class_data, mask):
     """Test fit for surface image."""
     warnings.simplefilter("ignore", ConvergenceWarning)
-    X, y = _make_surface_class_data()
+    X, y = _make_surface_class_data
     model = decoder(mask=mask)
     model.fit(X, y)
 
     assert model.coef_ is not None
 
 
+@pytest.mark.filterwarnings("ignore:After clustering and screening")
 @pytest.mark.parametrize("decoder", [_BaseDecoder, Decoder, DecoderRegressor])
 def test_decoder_fit_surface_with_mask_image(
-    _make_surface_class_data, decoder, mini_mask
+    _make_surface_class_data, decoder, surf_mask
 ):
     """Test fit for surface image."""
     warnings.simplefilter("ignore", ConvergenceWarning)
-    X, y = _make_surface_class_data()
-    model = decoder(mask=mini_mask)
+    X, y = _make_surface_class_data
+    mask = surf_mask()
```

Comment:
```suggestion
    mask = surf_mask()
```

no need to reassign the fixture